activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -138,7 +144,14 @@ def as_json(options = nil) # :nodoc:
 
 module Enumerable
   def as_json(options = nil) # :nodoc:
-    to_a.as_json(options)
+    respond_to = Kernel.instance_method(:respond_to?).bind(self)
```

Comment:
`Kernel.instance_method(:respond_to?)` instantiate a new `UnboundMethod` on each call, it could be constantized.

Then `bind(self)` instantiate a `Method`, you could use `bind_call` instead.

Reply:
thanks